src/client/components/vision/camera/view.tsx

```diff
@@ -0,0 +1,61 @@
+import { autorun } from 'mobx'
+import { action } from 'mobx'
+import { observer } from 'mobx-react'
+import * as React from 'react'
+import { Component } from 'react'
+
+import { CameraViewModel } from './view_model'
+
+@observer
+export class CameraView extends Component<{ viewModel: CameraViewModel }> {
+  private destroy: () => void = () => {}
+
+  componentDidMount() {
+    this.destroy = autorun(() => this.renderScene(), {
+      scheduler: requestAnimationFrame,
+    })
+  }
+
+  componentWillUnmount() {
+    this.destroy()
+  }
+
+  render() {
+
+    const { width, height } = this.props.viewModel
+
+    const aspectRatio = width / height
+    const percentage = 60
+
+    return (
+      <canvas style={{
+        display: 'inline',
+        width: `${percentage}vw`,
+        height: `${percentage / aspectRatio}vw`,
+        maxHeight: `${percentage}vh`,
+        maxWidth: `${percentage * aspectRatio}vh`,
+      }}
+        width={width}
+        height={height}
+        ref={this.onRef}/>
+    )
+  }
+
+  @action
+  private onRef = (canvas: HTMLCanvasElement | null) => {
+    this.props.viewModel.canvas = canvas
+  }
+
+  private renderScene() {
+    const { viewModel } = this.props
+    const { canvas, scene, camera } = viewModel
+    const renderer = viewModel.renderer(canvas)
+    if (renderer) {
+      renderer.render(scene, camera)
+
+      // This is needed as we constantly make new scene objects, if we don't do this the renderLists
+      // stores a reference to the scene and mesh causing a memory leak
+      renderer.renderLists.dispose()
```

Comment:
Do we need all the other dispose methods if we use this?

I'd like to know more about this, is this a bug/issue with three.js? Is there a ticket or three.js source code you could link to?

Reply:
I'm not 100% sure, and honestly this one is only here as it's the only way to dispose of scenes and cameras that are no longer used.
I'm actually trying to think of a good way to reuse the same scene object each time so I don't have to resort to this sledgehammer.